File: src/my_project/easy_problems/from201to250/check_if_word_occurs_as_prefix.py
from typing import List 

class Solution:
    def isPrefixOfWord(self, sentence: str, searchWord: str) -> int:
        
        list_sentence = sentence.split()
        len_sentence: int = len(list_sentence)
        len_search_word: int = len(searchWord)
        flag: bool = False 
        for i in range(len_sentence):
            flag = False 
            
            for j in range(len_search_word):
                if len(searchWord) <= len(list_sentence[i]):
                    if searchWord[j] != list_sentence[i][j]:
                        flag = True
                        continue
                else: 
                    flag = True
                    continue
            if flag == False: 
                return i + 1
        return -1
    
solution = Solution()

print(solution.isPrefixOfWord(sentence = "i love eating burger", searchWord = "burg"))

print(solution.isPrefixOfWord(sentence = "this problem is an easy problem", searchWord = "pro"))

print(solution.isPrefixOfWord(sentence = "i am tired", searchWord = "you"))

print(solution.isPrefixOfWord(sentence = "i use triple pillow", searchWord = "pill"))

print(solution.isPrefixOfWord(sentence = "hello from the other side", searchWord = "they"))

File: my_project_typescript/easy_problems_typescript/from101to150/checkIfWordOccursAsPrefix.ts
function isPrefixOfWord(sentence: string, searchWord: string): number {
    let listSentence: Array<String> = sentence.split(" ")
    let lenSentence: number = listSentence.length
    let lenSearchWord: number = searchWord.length
    let flag: boolean = false 

    for (let i = 0; i < lenSentence; i++){
        flag = false 
        for (let j = 0;  j < lenSearchWord; j++){
            if (lenSearchWord <= listSentence[i].length){
                if (searchWord[j] !== listSentence[i][j]){
                    flag = true 
                    continue 
                }
            }
            else{
                flag = true 
                continue
            }   
        }
        if (flag === false){
            return i + 1
        }
    }

    return -1
};


console.log(isPrefixOfWord("i love eating burger",  "burg"))

console.log(isPrefixOfWord("this problem is an easy problem",  "pro"))

console.log(isPrefixOfWord("i am tired",  "you"))

console.log(isPrefixOfWord("i use triple pillow",  "pill"))

console.log(isPrefixOfWord("hello from the other side",  "they"))

